src/Rotations.jl

```diff
@@ -0,0 +1,318 @@
+plane at `x` on the special orthogonal space `S=`$\mathrm{SO}(n)$ using the
+restriction of the metric from the embedding, i.e. $ (v,w)_x = v^\mathrm{T}w $.
+Tangent vectors are represented by matrices.
+"""
+inner(S::Rotations, x, w, v) = dot(w, v)
+
+project_tangent!(S::Rotations, w, x, v) = w .= (transpose(v).-v)./2
+
+function exp!(S::Rotations, y, x, v)
+    y .= x * exp(v)
+    return y
+end
+
+@doc doc"""
+    log!(M, v, x, y)
+
+compute the logarithmic map on the [`Rotations`](@ref) manifold
+`M`$=\mathrm{SO}(n)$, which is given by
+```math
+\operatorname{log}_{x} y =
+  \frac{1}{2} \bigl(\operatorname{Log}(x^{\mathrm{T}}y)
+  - (\operatorname{Log} x^{\mathrm{T}}y)^{\mathrm{T}}),
+```
+where $\operatorname{Log}$ denotes the matrix logarithm
+and save the result to `v`.
+"""
+function log!(S::Rotations, v::TV, x, y) where TV
+    U = transpose(x) * y
+    if (x != y) && (abs.(U) == one(U))
+        throw( ErrorException("The points $x and $y are antipodal, thus these input parameters are invalid.") )
+    end
+    # MMatrix doesn't have `log` defined
+    U1 = TV(real(log(Array(U))))
+    v .= (U1 .- transpose(U1))./2
+    return v
+end
+
+injectivity_radius(M::Rotations, x) = π*sqrt(2.0)
+
+@doc doc"""
+    PolarRetraction
+
+Retraction on the rotations manifold using the polar method.
+This SVD-based retraction is a second-order approximation of the
+exponential map. Let
+
+$USV = x + xv$
+
+be the singular value decomposition, then the formula reads
+
+$\operatorname{retr}_x v = UV^\mathrm{T}$
+
+Retraction is performed by the function [`retract!(::Rotations, y, x, v, ::PolarRetraction)`](@ref)
+"""
+struct PolarRetraction <: AbstractRetractionMethod end
+
+@doc doc"""
+    retract_polar!(M::Rotations, y, x, v, method::PolarRetraction)
+
+Compute the SVD-based retraction [`PolarRetraction`](@ref), a second-order
+approximation of the exponential map.
+"""
+function retract!(M::Rotations, y, x, v, method::PolarRetraction)
+    A = x + x*v
+    S = svd(A)
+    y .= S.U * transpose(S.V)
+    return y
+end
+
+"""
+    QRRetraction
+
+Retraction on the rotations manifold using the QR method.
+"""
+struct QRRetraction <: AbstractRetractionMethod end
+
+@doc doc"""
+    retract!(M, y, x, v, method::QRRetraction)
+
+This QR-based retraction is a first-order approximation of the exponential map.
+"""
+function retract!(M::Rotations, y::AbstractArray{T}, x, v, method::QRRetraction) where T
+    A = x + x*v
+    qr_decomp = qr(A)
+    d = diag(qr_decomp.R)
+    D = Diagonal( sign.(d .+ convert(T, 0.5)) )
+    y .= qr_decomp.Q * D
+    return y
+end
+
+retract!(M::Rotations, y, x, v) = retract!(M, y, x, v, QRRetraction())
+
+"""
+    PolarInverseRetraction
+
+Inverse retraction on the rotations manifold using the polar method.
+"""
+struct PolarInverseRetraction <: AbstractInverseRetractionMethod end
+
+injectivity_radius(::Rotations, x, ::PolarInverseRetraction) = π*sqrt(2.0)/2
```

Comment:
I think we should update our documentation of `injectivity_radius` to clarify that we're not using the technical definition of "injectivity radius" (maximum radius of the geodesic ball over which the exponential map is diffeomorphic) and instead using "maximum radius over which the specified retraction is injective (has a left inverse)"

Reply:
I've edited the docstring for `injectivity_radius` that operates on inverse retraction objects (in `ManifoldMuseum.jl`)